### ISSUE TITLE
ch9 code clarification on applying softmax

### ISSUE THREAD

**Issue by kimsjune** [StatQuest/signa#26]
Hi there,

Awesome book and code! 

I'm seeing a contradiction in comments below:

```python
        ## Now we need to concatenate the 4 output tensors so that we can run them through
        ## the SoftMax function. However, because they are tensors (and have gradients attached to them),
        ## we can't just combine them in a simple list like this...
        # output_values = [output1, output2, output3, output4] ## THIS WILL NOT WORK
        ## ...because that would strip off the gradients.
        ## Instead, we use torch.stack(), which retains the gradients.
        output_presoftmax = torch.stack([output1, output2, output3, output4])
        ## NOTE: The the loss function we are using, nn.CrossEntropyLoss, automatically applies softmax for us, so we
        ##       need to do that ourselves. If we want to actually use this network to predict the next word
        ##       (instead of just using it for the Word Embedding values), then we'll need to apply the softmax() function
        ##       ourselves (or just look to see what output value is largest).
```

"nn.CrossEntropyLoss automatically applies softmax for us, so we need to do that ourselves"? If it is automatic, what are we exactly ourselves?

Later on, there's the following comment and code:

```python
## Let's see what the model predicts

## First, let's create a softmax object...
softmax = nn.Softmax(dim=0) ## dim=0 applies softmax to rows, dim=1 applies softmax to columns
```
Why are we creating a softmax object when "nn.CrossEntropyLoss automatically applies softmax for us"? Indeed, `nn.Softmax(dim=0)` is required to get the final predictions (next words).

I'd appreciate it if anyone could clarify! Thank you.

Environment:

pytorch-lightning==2.5.6
segmentation_models_pytorch==0.5.0
torch==2.9.1
torchmetrics==1.8.2
torchvision==0.24.1
